src/django_components/util/command.py

```diff
@@ -0,0 +1,407 @@
+    """
+    The basic type of action to be taken when this argument is encountered at the command line.
+    """
+    dest: Optional[str] = None
+    """
+    Name of the attribute under which sub-command name will be stored; by default `None`
+    and no value is stored.
+    """
+    required: Optional[bool] = None
+    """
+    Whether or not a subcommand must be provided, by default `False` (added in 3.7)
+    """
+    help: Optional[str] = None
+    """
+    Help for sub-parser group in help output, by default `None`.
+    """
+    metavar: Optional[str] = None
+    """
+    String presenting available subcommands in help; by default it is None and
+    presents subcommands in form `{cmd1, cmd2, ..}`.
+    """
+
+    def asdict(self) -> dict:
+        """Convert the dataclass to a dictionary, stripping out fields with `None` values"""
+        return _remove_none_values(asdict(self))
+
+
+@dataclass
+class CommandParserInput:
+    """
+    Typing for the input to the
+    [`ArgumentParser`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser)
+    constructor.
+    """
+
+    prog: Optional[str] = None
+    """The name of the program (default: `os.path.basename(sys.argv[0])`)"""
+    usage: Optional[str] = None
+    """The string describing the program usage (default: generated from arguments added to parser)"""
+    description: Optional[str] = None
+    """Text to display before the argument help (by default, no text)"""
+    epilog: Optional[str] = None
+    """Text to display after the argument help (by default, no text)"""
+    parents: Optional[Sequence[ArgumentParser]] = None
+    """A list of ArgumentParser objects whose arguments should also be included"""
+    formatter_class: Optional[Type["_FormatterClass"]] = None
+    """A class for customizing the help output"""
+    prefix_chars: Optional[str] = None
+    """The set of characters that prefix optional arguments (default: ‘-‘)"""
+    fromfile_prefix_chars: Optional[str] = None
+    """The set of characters that prefix files from which additional arguments should be read (default: `None`)"""
+    argument_default: Optional[Any] = None
+    """The global default value for arguments (default: `None`)"""
+    conflict_handler: Optional[str] = None
+    """The strategy for resolving conflicting optionals (usually unnecessary)"""
+    add_help: Optional[bool] = None
+    """Add a -h/--help option to the parser (default: `True`)"""
+    allow_abbrev: Optional[bool] = None
+    """Allows long options to be abbreviated if the abbreviation is unambiguous. (default: `True`)"""
+    exit_on_error: Optional[bool] = None
+    """Determines whether or not ArgumentParser exits with error info when an error occurs. (default: `True`)"""
+
+    def asdict(self) -> dict:
+        """Convert the dataclass to a dictionary, stripping out fields with `None` values"""
+        return _remove_none_values(asdict(self))
+
+
+#############################
+# Command logic
```

Comment:
The rest of the file the is about:
- the `ComponentCommand` class, the class for defining commands for extensions
- helpers for transforming the declarative command definition into what `argparse` expects.